react-table-config.d.ts

```diff
@@ -0,0 +1,50 @@
+declare module 'react-table' {
+  export interface TableOptions<D extends object>
+    extends UseExpandedOptions<D>,
+      UseFiltersOptions<D>,
+      UseGroupByOptions<D>,
+      UsePaginationOptions<D>,
+      UseRowSelectOptions<D>,
+      UseSortByOptions<D>,
+      UseFiltersOptions<D> {}
+
+  export interface TableInstance<D extends object = {}>
+    extends UseColumnOrderInstanceProps<D>,
+      UseExpandedInstanceProps<D>,
+      UseFiltersInstanceProps<D>,
+      UseGroupByInstanceProps<D>,
+      UsePaginationInstanceProps<D>,
+      UseRowSelectInstanceProps<D>,
+      UseRowStateInstanceProps<D>,
+      UseSortByInstanceProps<D> {}
+
+  export interface TableState<D extends object = {}>
+    extends UseColumnOrderState<D>,
+      UseExpandedState<D>,
+      UseFiltersState<D>,
+      UseGroupByState<D>,
+      UsePaginationState<D>,
+      UseRowSelectState<D>,
+      UseSortByState<D> {}
+
+  export interface Column<D extends object = {}>
+    extends UseFiltersColumnOptions<D>,
+      UseGroupByColumnOptions<D>,
+      UseSortByColumnOptions<D> {}
+
+  export interface ColumnInstance<D extends object = {}>
+    extends UseFiltersColumnProps<D>,
+      UseGroupByColumnProps<D>,
+      UseSortByColumnProps<D> {}
+
+  export interface Cell<D extends object = {}>
+    extends UseTableCellProps<D>,
+      UseGroupByCellProps<D>,
+      UseRowStateCellProps<D> {}
+
+  export interface Row<D extends object = {}>
```

Comment:
I had to add `Omit<UseTableRowProps<D>, 'state'>` for I'm using `getRowProps` and `cells` on Row. Also had to omit `state: object` because it's already in `UseRowStateRowProps` as `state: UseRowStateLocalState<D>`.

```suggestion
  export interface Row<D extends object = {}> extends UseExpandedRowProps<D>,
    UseGroupByRowProps<D>,
    UseRowSelectRowProps<D>,
    Omit<UseTableRowProps<D>, 'state'>,
    UseRowStateRowProps<D> {
  }
```

Reply:
I noticed this as well, and raised this on https://github.com/tannerlinsley/react-table/pull/1535, though I'm failing to find the comment right now.  I'll add it to the list again.

Reply:
Cool. Everything else works for me (though I have only one table for v7). Thanks!

Reply:
`state` only appears to be on `row` if you use `useRowState` plugin hook. In which, it forks the key `rowState` out of `state`.